spec/spec/have_api_error_spec.rb

```diff
@@ -60,10 +61,31 @@
   context "expected status is specified" do
     context "status is the expected status" do
       let(:response_code) { 404 }
-      it "indicates there is an error" do
-        expect(response).to have_api_error(status: 404,
-                                           code: errors.first[:code],
-                                           message: errors.first[:message])
+      context "there is an error" do
+        it "indicates there is an error" do
+          expect(response).to have_api_error(status: 404,
+                                             code: errors.first[:code],
+                                             message: errors.first[:message])
+        end
+      end
+      context "there is not an error" do
+        context "and the status code is what's wrong" do
+          let(:response) {
+            double(
+              response_code: 401,
+              body: { errors: errors }.to_json)
+          }
+          it "fails the test" do
+            begin
+              expect(response).to have_api_error(status: 404,
+                                                 code: errors.first[:code],
+                                                 message: errors.first[:message])
+            rescue Exception => ex
```

Comment:
I'm not sure if there's a better way to test that a matcher fails in the right way.